torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -24,6 +24,30 @@ namespace jit {
 
 namespace {
 
+void replaceConv1dWithConv2d(std::shared_ptr<Graph>& graph) {
+  std::string conv_1d_pattern = R"(
+    graph(%input, %weight, %bias, %stride:int[], %padding:int[], %dilation:int[], %groups:int):
+        %r = aten::conv1d(%input, %weight, %bias, %stride, %padding, %dilation, %groups)
+        return (%r) )";
+
+  std::string conv_2d_pattern = R"(
+    graph(%input, %weight, %bias, %stride:int[], %padding:int[], %dilation:int[], %groups:int):
+        %padding_w : int = prim::ListUnpack(%padding)
```

Comment:
Why only padding is adjusted like this? What about stride, and dilations?

Reply:
I think we can handle stride and dilation similarly as well. However, it might not be needed since we only move in one dimension in conv1d, so the added dimension for stride and dilation may not matter too much. Padding is different, because we don't want to pad the added dimension.

Reply:
I think we should be consistent and clear here. It is ok that conv2d might handle that gracefully, but it may change and not handle undefined value gracefully. We should avoid baking assumption about what some function might do, unless it is strongly coupled with our implementation.